### DISCUSSION TITLE
Configure dashboards via code (IaC)

### DISCUSSION THREAD

**Discussion by Asgoret** [grafana/grafana#44031] [Dashboards, Panels & Visualizations]
Hi to all.
When does grafana publish and will continue to support and develop tool which will configure dashboards through code?
At FOSDEM 2020 (it was in grafana blogs too([link](https://grafana.com/blog/2020/02/26/how-to-configure-grafana-as-code/)))you talk about [grafonnet](https://github.com/grafana/grafonnet-lib). This library has lack documentation, example and other helpers. But it's work. At some point, grafana abandoned it because they release grafana 8 and the library was still for grafana 7 (and in 2022 nothing changes). They opened a discussion [about the future of grafonnet](https://github.com/grafana/grafonnet-lib/issues/332), then published some [document](https://docs.google.com/document/d/1qp3flLb6q1dPqTmHQ6MV-o6lOmlctfMNpzTx2MdzucY/edit#heading=h.pnr4jkg68wb6) about templating at CUE.....and abandoned it too in Feb of 2021;D Moved to the separate repository for [templating at CUE](https://github.com/grafana/thema). But it's *pre-alpha*...

So...since 2020 nothing changes) So, my question is: When grafana will publish, develop and support on long term technology of configuration dashboards via code?

**Comment by selmison** [comment]
@sdboyer 

Recently in this PR (#47653) you introduced the `Coremodels framework`. And this PR (#44840) mentions that the goal of the Coremodels is to be a **single source of truth** of the Grafana objects.

Can we begin to use this "framework" to replace the Grafonnet as tool to describe dashboards through code? It seems like this [Lineage](https://github.com/grafana/grafana/blob/main/pkg/coremodel/dashboard/coremodel.cue) is a good starting point.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
